### DISCUSSION TITLE
Add hideEmptyState option to Table

### DISCUSSION THREAD

**Discussion by Tray2** [filamentphp/filament#9577] [Ideas]
When you have an empty table in your application, you see a big fat "No <model>" and an X-icon, in some cases it would be good to be able to hide the table completely.

I recently came across a need for this functionality, and had to resort to putting an if statement around the `$this->table`.
I know that you can set the empty state header, description and icon to make it appear as hidden, but I think it would be even greater if you could just chain a `->hideEmptyState()` to the table.

Like this:

```
public function table(Table $table): Table
    {
        return $table
            ->query(OpenHours::query()
                ->whereNotNull('date')
                ->orderBy('date'))
            ->columns([
                TextColumn::make('date')
                    ->label('Deviating Dates')
                    ->formatStateUsing(function (OpenHours $record) {
                        $date = date_create($record->date);
                        return date_format($date, 'M dS');
                    }),
                TextColumn::make('open_hour')
                    ->label(''),
                TextColumn::make('')
                    ->label('')
                    ->default('-'),
                TextColumn::make('close_hour')
                    ->label(''),
            ])->paginated(false)
            ->hideEmptyState();
    }
```

Or another option would be to set it in the livewire component like so.

```
<div>
    {{ $this->table->hideIfEmpty() }}
</div>
```

For now I made this workaround

```
<div>
    @if($this->table->getRecords()->isNotEmpty())
        {{ $this->table }}
    @endif
</div>
```

**Comment by lartisan** [comment]
Nice, but this didn't work for me when having polling on the table 😬
